File: app/src/main/java/com/hjc/base/ui/video/adapter/MyFragmentPagerAdapter.java
package com.hjc.base.ui.video.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.List;

public class MyFragmentPagerAdapter extends FragmentStatePagerAdapter {

    private final List<Fragment> mFragments;

    public MyFragmentPagerAdapter(@NonNull FragmentManager fm, int behavior, List<Fragment> fragments) {
        super(fm, behavior);
        this.mFragments = fragments;
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return mFragments.get(position);
    }

    @Override
    public int getCount() {
        return mFragments.size();
    }
}

File: module-other/src/main/java/com/hjc/module_other/adapter/MyFragmentPagerAdapter.kt
package com.hjc.module_other.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

/**
 * @Author: HJC
 * @Date: 2021/9/26 16:36
 * @Description: ViewPager Adapter
 */
class MyFragmentPagerAdapter(fm: FragmentManager, behavior: Int, private val mFragments: List<Fragment>) :
    FragmentStatePagerAdapter(fm, behavior) {

    override fun getItem(position: Int): Fragment {
        return mFragments[position]
    }

    override fun getCount(): Int {
        return mFragments.size
    }
}